tests/crawler/test_crawler_unittest.py

```diff
@@ -28,3 +33,52 @@ def test_close_all_windows_except_first(self):
         crawler.close_all_windows_except_first(self.driver)
 
         assert (len(self.driver.window_handles) == 1), 'some windows were not closed properly.'
+
+
+class TestCrawlerLive(unittest.TestCase):
+    SERVER_SETUP_TRIES = 10
+
+    @classmethod
+    def setUpClass(cls):
+        cls.server = multiprocessing.Process(target=run_server)
+        cls.server.start()
+        test_driver = webdriver.Firefox(executable_path=GeckoDriverManager().install())
+        for try_id in range(cls.SERVER_SETUP_TRIES):
+            try:
+                print(f'class setup tries: {try_id}/{cls.SERVER_SETUP_TRIES}')
+                test_driver.get(website_app.WEBSITE_URL)
+                assert test_driver.title == WEBSITE_TITLE
+                test_driver.quit()
+                return
+            except WebDriverException as e:
+                print('got exception:', e)
```

Comment:
on failure, we should wait a bit before the next retry (in this case waiting for 1 sec is fine)

Reply:
`test_driver.get()` takes about 5s before timeout, so an extra sleep is a waste of time

Reply:
OK, sounds good then.